checks/testdata/Dockerfile-curl-file-sh

```diff
@@ -0,0 +1,33 @@
+
+# Copyright 2021 Security Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this exe except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM python:3.7@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2
+
+RUN curl exe 2>&1 > /tmp/exe1 && /tmp/exe1
+RUN curl exe2 2>&1 > /tmp/exe2 ; /tmp/exe2
+RUN curl exe2 2>&1 > /tmp/exe4 ; /tmp/exe4
+RUN curl exe2 2>&1 > /tmp/exe3 ; \
+    /tmp/exe3
+
+RUN curl file 2>&1 > /tmp/file1 && sh /tmp/file1
+RUN curl file2 2>&1 > /tmp/file2 ; sh /tmp/file2
+RUN curl file2 2>&1 > /tmp/file4 ; sh /tmp/file4
+RUN curl file2 2>&1 > /tmp/file3 ; \
+    bash /tmp/file3
+
+RUN ["curl", "file", "2>&1", ">", "/tmp/file111", "&&", "sh", "/tmp/file111"]
```

Comment:
Per https://docs.docker.com/engine/reference/builder/#run

I don't think this is a correct example. This form does not invoke a shell by default, so we'd want to look for something like:

```
RUN ["/bin/bash", "-c", "CHECK THIS"] 
```

instead.

This second form does not invoke a shell by default.

Reply:
I see what you mean.
The command example would be invalid and would not work in practice, right? (since redirection implies a shell is running).

I'll update the example. Note that the second form is detected with the current implementation. I simply `Join` the array to a string and give it to the parser. As far as the check is concerned, `RUN []` and `RUN string` are equivalent.

I've taken some other shortcuts in the implementation: it will detect things that can never be valid, but this allowed me to simplify the implementation. Example: `wget <> | bash` would not work because wget saves to a file, yet will be detected by the current implementation. I don't think this should be a problem since the example above should never appear in real scripts.

Let me know what you think though. I can refine things if needed.

Reply:
actually, I was wrong: Im not checking for `bash -c ""` at all. I'll send you a followup PR

Reply:
see https://github.com/ossf/scorecard/pull/600